consent/api/v1/views.py

```diff
@@ -187,19 +214,40 @@ def post(self, request):
 
         """
         try:
+            # Ensure that we got the required explicit parameters in the API
             username, course_id, enterprise_customer_uuid = self.get_required_query_params(request)
         except ConsentAPIRequestError as invalid_request:
             return Response({'error': str(invalid_request)}, status=HTTP_400_BAD_REQUEST)
 
+        # Determine whether consent is needed and if it's been provided already.
         user_id = get_user_id(username)
-        self.set_consent_state(True, user_id, course_id, enterprise_customer_uuid)
+        exists = consent_exists(user_id, course_id, enterprise_customer_uuid)
         provided = consent_provided(user_id, course_id, enterprise_customer_uuid)
-        required = consent_required(user_id, course_id, enterprise_customer_uuid)
+
+        # We can never require consent for a course/EC combo if the course isn't
+        # in the Enterprise's course catalog.
+        enterprise_customer = get_enterprise_customer(enterprise_customer_uuid)
+        if enterprise_customer and not enterprise_customer.catalog_contains_course_run(request.user, course_id):
+            required = False
+        else:
+            required = consent_required(user_id, course_id, enterprise_customer_uuid)
```

Comment:
Again, if we can move this inside `consent_required` we don't have to duplicate this code here.